### ISSUE TITLE
Add FrontendRegion, changing Frontend and PrettyScreen traits

### ISSUE THREAD

**Issue by e-matteson** [justinpombrio/synless#12]
Paste notes in a comment?

**Comment by justinpombrio** [comment]
```
struct FrontEndScreen<'a, T: FrontEnd> {
    frontend: &'a mut T,
    region: Region
}

trait Frontend {
    fn print();
    fn get_screen<'a>(region: &'a mut Region) -> FrontendScreen<'a, Self>;
}

impl<'a, T> PrettyScreen for FrontEndScreen<'a, T> {
    fn print(&mut self, pos) {
        self.frontend.print(OFFSET(self.region, pos));
    }
}
```

**Comment by justinpombrio** [comment]
(Should probably call it a `FrontendScreen` rather than a `FrontendRegion`, since the point is that it implements `PrettyScreen`.)

**Comment by e-matteson** [comment]
Closed by #19, though with different design and names.
